### DISCUSSION TITLE
in lazer editor please show what the shortcuts actually are in the ribbon menu

### DISCUSSION THREAD

**Discussion by silk-hyacinth** [ppy/osu#29134] [Ideas] [answered]
Hey so in lazer currently it is really unintuitive and difficult to get started or adjust from stable editor right now since it is impossible or super unintuitive to find 90% of the shortcuts that as a mapper you use a ton. 

The stable editor right now is really well designed for the user experience since instantly on logging in the ribbon menu at the top clearly shows all the tools available in the editor and it isn't hidden behind 3 right click menus like the scale and rotate tools are currently in lazer and in the edit menu that is very clearly visible instantly upon loading into the editor it is simple to find what all the shortcuts are that are necessary for the mapping workflow.
  
![Screenshot 2024-07-27 041548](https://github.com/user-attachments/assets/04f51c00-fb3f-4787-b92f-237d79996f88)
as you can see here in the stable editor it is super straightforward and intuitive to find all the functionality of the editor in a single menu,

whereas in lazer it is currently obfuscated and scattered in multiple random places around the screen; it makes much more sense to put all the editing tools in one menu but now there are some in the edit ribbon at the top but half of the ones previously in the edit ribbon are now moved to another ribbon on the right side, and for some reason after 10 minutes of searching I still can not find a place in the editor that says what the shortcuts for rotate, scale, polygon circle, slider to stream, et cetera are in the editor (i can't even figure out if old ctrl+shift+d even exists..?). Currently lazer's ribbon has for some reason nothing in it? like see here 
![image](https://github.com/user-attachments/assets/c28123ae-f34d-440e-bd12-b41b4879d959) there are six items in here, zero shortcuts listed. Please add to this menu all of the functionalities that exist in lazer's editor and all of their respective shortcuts like stable's editor's menu has.

tl;dr 
1. please don't have everything scattered everywhere and then not be accessible in the main ribbon menus on the top where a user normally expects to find all the functionality; if you put stuff in the right side menu at least also put it in the ribbon at the top (this is done in stable for example, grid snap is in both the ribbon and the top) so that I don't have to go on a scavenger hunt on my screen to find what functionality actually exists in the lazer editor. all functionalities should be listed in one place.
2. please PLEASE in the top menu show what the keyboard shortcuts are for all (ALL!!!!) functionalities that exist in the editor just like it is done in stable. ux >>>>> ui. this is an editor, not a logo.

thank you for listening to my ted talk.

**Comment by Joehuu** [comment, marked as answer]
This is already in the works, see https://discord.com/channels/188630481301012481/188630652340404224/1263442822300827719.

**Reply by silk-hyacinth** [reply]
Yes that is very cool but also please dump all shortcuts that exist into this menu so that the menu is not just copy and paste and redo.

(ctrl-r, whatever the shortcut is for old control-shift-s is now, slider to stream, et cetera)

thank you
